### ISSUE TITLE
HELP: Deleted all channels on my hub

### ISSUE THREAD

**Issue by jeroenpraat** [redmatrix/redmatrix#265]
_From @jeroenpraat on October 27, 2014 22:1_

I opened by accident all the channels on the channel admin page on my hub, incl. the delete button. Now they are all gone. I have a a DB dump from last Thursday. Should I restore that? Or are there other ways to fix this extremely STUPID error I made?

_Copied from original issue: friendica/red#667_

**Comment by jeroenpraat** [comment]
DAMN how could I be so stupid!!!

**Comment by jeroenpraat** [comment]
_From @friendica on October 27, 2014 22:15_

You'll need the backup. Deleting channels is pretty destructive.

**Comment by jeroenpraat** [comment]
OK, trying to do that :(

**Comment by jeroenpraat** [comment]
_From @ghost on October 27, 2014 22:18_

Unfortunately, it looks like account_remove() works quite well...

Restoring your dump is the best bet - even if you have clones, it's a simpler recovery from a restored database than from nothing at all.

**Comment by jeroenpraat** [comment]
_From @ghost on October 27, 2014 22:18_

Oh.  Three comments while I was writing that.  Live updates don't work in Konqueror

**Comment by jeroenpraat** [comment]
_From @friendica on October 27, 2014 22:19_

I'm not sure how you would've missed this popup:

    Selected channels will be deleted!
    Everything that was posted in these channels on this site will be permanently deleted!
    Are you sure?

but let's just worry about getting back to normal. Then figure out if there's a better way to prevent this from happening.

**Comment by jeroenpraat** [comment]
The accounts seems to be unaffected. I tried to change the channel and xchan tables, but no effect. I will restore it.

@friendica I wanted to check the channels on my hub to see if there were no offending channels. I selected the whole list, but also the red delete buttons. Than I clicked 'open all links in new tabs'.  Before I noticed it, all channels where gone.

**Comment by jeroenpraat** [comment]
DB dump successfully restored. Now polling.

**Comment by jeroenpraat** [comment]
All seems fine. Posts from others are back. Own posts are gone after October 23th. Same for other local channels. Posts posted to forum channels are still there. Also comments. Yeaah.

**Comment by jeroenpraat** [comment]
_From @ghost on October 27, 2014 23:32_

jeroenpraat, opensource, kas, rms, ham, anarchism, android, gif, and mapporn are all still deleted.  This will fix itself, but you might want to do a refresh all to speed things up if any of these are important.

**Comment by jeroenpraat** [comment]
I'm checking them now one for one, but they all are there. Mapporn misses that cheese map of Britain, but it is on the cheese forum (that one is not mine), so it wil probably come back. Also that Piri Map is not there (posted by Zottel if I remember correctly). RMS is also there. Last post is from 6 days ago. All channels seems fine, with a few external posts still missing, but that will probably fix itself.

What do you mean with a refresh?

**Comment by jeroenpraat** [comment]
_From @ghost on October 27, 2014 23:46_

"Deleted" as in "deleted on the network".  They have XCHAN_FLAGS_DELETED.  On the plus side, this is just a flag, and can be changed.  Just update your profile (you don't need to actually change anything), or perform some other action to update the directory, and this will go out to all your connections too, as long as your flags are correct at the original (which they will be now, since you restored a backup).

**Comment by jeroenpraat** [comment]
Ok, I get it. Updating the profiles of those channels you mentioned I presume?

**Comment by jeroenpraat** [comment]
Done, except opensource, kas and ham, cause they are not mine.

**Comment by jeroenpraat** [comment]
I've checked why this could happen. Clicking on an X after a channel name on /admin/channels/ will delete that channel, but only after a dialog window asks for confirmation. BUT, when you open that link in a new tab with the context menu (in Firefox), there is no confirmation dialog window at all. Doing that for all channels (what I did with selecting all links, incl. the X links by accident) will delete all channels.

I'm closing this now. Thanks for your help.
